File: backend/Domain/Entities/Hotel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;

namespace Domain.Entities
{
    public class Hotel : BaseEntity
    {
        [Range(1, 5)]
        public int StarRating { get; set; }
        public bool RestaurantOnSite { get; set; }
    }
}

File: mobile/lib/src/features/dashboard/domain/entities/hotel.dart
import 'package:equatable/equatable.dart';

class Hotel extends Equatable {
  final int starRating;
  final bool restaurantOnSite;

  const Hotel({
    required this.starRating,
    required this.restaurantOnSite,
  });

  @override
  List<Object?> get props => [starRating, restaurantOnSite];
}
